lib/internal/Magento/Framework/App/DeploymentConfig/Writer/PhpFormatter.php

```diff
@@ -51,18 +53,42 @@ private function formatData($data, $comments = [], $prefix = '  ')
                     $elements[] = $prefix . " */";
                 }
 
-                $elements[] = $prefix . var_export($key, true) . ' => ' .
-                    (!is_array($value) ? var_export($value, true) . ',' : '');
+                $elements[] = $prefix . $this->varExportShort($key) . ' => ' .
+                    (!is_array($value) ? $this->varExportShort($value) . ',' : '');
 
                 if (is_array($value)) {
-                    $elements[] = $prefix . 'array (';
+                    $elements[] = $prefix . '[';
                     $elements[] = $this->formatData($value, [], '  ' . $prefix);
-                    $elements[] = $prefix . '),';
+                    $elements[] = $prefix . '],';
                 }
             }
             return implode("\n", $elements);
         }
 
         return var_export($data, true);
     }
+
+    /**
+     * If variable to export is an array, format with the php >= 5.4 short array syntax. Otherwise use
+     * default var_export functionality.
+     *
+     * @param mixed   $var
+     * @param integer $depth
+     * @return string
+     */
+    private function varExportShort($var, $depth = 0)
+    {
+        if (gettype($var) === 'array') {
```

Comment:
Is there any specific reason not to use `is_array`?
Additionally, readability may be slightly improved by reversing the logic like

```php
if (!is_array($var)) {
    return var_export($var, true);
}
/// Transforming to short array syntax here
...
```